sdk/python/kfp/dsl/dsl_utils.py

```diff
@@ -14,13 +14,24 @@
 """Utilities functions KFP DSL."""
 
 import re
-from typing import Union
+from typing import Callable, List, Optional, Union
+
+from kfp.components import _structures
 from kfp.pipeline_spec import pipeline_spec_pb2
 
 _COMPONENT_NAME_PREFIX = 'comp-'
 _TASK_NAME_PREFIX = 'task-'
 _EXECUTOR_LABEL_PREFIX = 'exec-'
 
+# TODO: Support all declared types in
+# components._structures.CommandlineArgumenType
+_CommandlineArgumentType = Union[str, int, float,
```

Comment:
Since this is being used from within pipeline dsl, can we let users use InputValue, InputPath, etc (from python function-based components)?

Reply:
Took a look, actually they're for different use cases per current implementation. The ones from `_python_op.py` are used as type annotation, the value passed to the ctor is a type:
https://github.com/kubeflow/pipelines/blob/eb00b7b561e4e882e634daa072a06915bfd16f55/sdk/python/kfp/components/_python_op.py#L54-L57

Whereas the ones used here are for placeholder replacement, the value passed to the ctor is the name of an input/output:
https://github.com/kubeflow/pipelines/blob/eb00b7b561e4e882e634daa072a06915bfd16f55/sdk/python/kfp/components/_structures.py#L115-L124

Reply:
Yes. These are intentionally separate. `kfp.components.structures.InputPathPlaceholder(input_name)` is used when describing a container component implementation. `kfp.components.InputPath(input_type)` annotation class is only for Lightweight python components, has more high-level semantics and affects both component interface (input type) and implementation (code generation and the usage of the inputPath placeholder).